lib/grpc/telemetry/client.ex

```diff
@@ -0,0 +1,36 @@
+defmodule GRPC.Telemetry.Client do
+  @moduledoc """
+  Telemetry interceptor for GRPC client.
```

Comment:
I don't think we should go the interceptor route for telemetry. I think we should add the span as a wrapper for the actual adapter.send_request call.

Also, perhaps we should use execute instead of span, so we can add :headers_received and other intermediate steps instead of just :start and :stop

Reply:
I see. Let’s try

Reply:
> I don't think we should go the interceptor route for telemetry. I think we should add the span as a wrapper for the actual adapter.send_request call.
> 
> Also, perhaps we should use execute instead of span, so we can add :headers_received and other intermediate steps instead of just :start and :stop

What kind of value you expected from :headers_received? Is it headers when call recv_header from adapter?

Reply:
We can add the time that a given request took so far. If we introduce the request id as a first-class metadatum for the metrics, a monitoring system can easily show how much time it took to process each step (before headers, processing headers, total time since the beginning...)